tools/tsp-client/src/index.ts

```diff
@@ -0,0 +1,104 @@
+import * as path from "node:path";
+
+import { createPackageJson, installDependencies } from "./npm.js";
+import { createTempDirectory, removeDirectory, tryReadTspLocation, writeFileTree } from "./fs.js";
+import { Logger, printBanner, enableDebug, printVersion } from "./log.js";
+import { doesFileExist, downloadTsp, isValidUrl, rewriteGitHubUrl } from "./network.js";
+import { compileTsp } from "./typespec.js";
+import { getEmitterPackage } from "./languageSettings.js";
+import { getOptions } from "./options.js";
+
+async function discoverMainFileUrl(outputDir: string): Promise<string> {
+  const tspUrl = await tryReadTspLocation(outputDir);
+  if (!tspUrl) {
+    throw new Error("No tsp-location.yaml found, and no main file specified");
+  }
+  Logger.debug(`Using base url from  tsp-location.yaml: ${tspUrl}`);
+  for (const mainFile of ["client.tsp", "main.tsp"]) {
+    let url = new URL(mainFile, tspUrl).toString();
+    Logger.debug(`Checking for file: ${url}`);
+    if (await doesFileExist(url)) {
+      Logger.debug(`Found main file: ${url}`);
+      return url;
+    }
+  }
+  throw new Error(`No main.tsp or client.tsp found for base url ${tspUrl}`);
+}
+
+async function downloadAndCompile({
+  rootUrl,
+  emitter,
+  noCleanup,
+  emitterOutputPath,
+}: {
+  rootUrl: string;
+  emitter: string;
+  noCleanup: boolean;
+  emitterOutputPath: string;
+}) {
+  const { moduleImports, fileTree } = await downloadTsp(rootUrl);
```

Comment:
We should probably support a mode that can work against a local spec clone as well to by-pass the need to download any files.

Reply:
Yeah! We could either copy from a local location or even work out of it if we weren't worried about polluting the source with `node_modules` folders. It might be a nice way to bootstrap a "tweak the TSP and iterate" workflow.